### PR TITLE
use pixman for cursor plane rendering

### PR DESCRIPTION
**Pull request by cmeissl**
update necessary after https://github.com/Smithay/smithay/pull/1294 to use cursor plane for default cursor when rendering is needed

### CODE REVIEW

**Review by YaLTeR — COMMENTED** [review]
Gave this a quick test, seems to work nicely, and `render_cursor_plane` in Tracy went from ~1.5 ms mean to ~0.4 ms mean, cool!

File: src/cursor.rs [hunk 1]
```diff
@@ -254,24 +252,14 @@ impl CursorTextureCache {
                     .map(|frame| {
                         let _span = tracy_client::span!("create TextureBuffer");
```
[the remark this answers is not in the document]
Comment (cmeissl) [reply src/cursor.rs]: true, done

File: src/niri.rs [hunk 2]
```diff
@@ -1670,26 +1670,18 @@ impl Niri {
...
-                        PrimaryGpuTextureRenderElement(TextureRenderElement::from_texture_buffer(
-                            pointer_pos.to_f64(),
-                            &texture,
-                            None,
-                            None,
-                            None,
-                            Kind::Cursor,
-                        )),
-                    ));
+                if let Ok(element) = MemoryRenderBufferRenderElement::from_buffer(
```
[the remark this answers is not in the document]
Comment (cmeissl) [reply src/niri.rs]: sure, done

**Comment by cmeissl** [comment]
> Gave this a quick test, seems to work nicely, and `render_cursor_plane` in Tracy went from ~1.5 ms mean to ~0.4 ms mean, cool!

This (and the comment in the smithay PR from drakulix about the format), reminded me that I forgot to check the
format in niri. So this now also includes the format correction. This has two nice effects, first the color (if you have colored
cursors) is now correct and it also enables the fast-path where we skip rendering all together if the size matches (no scaling or transform). So I would expect that if the cursor size matches the physical size it should take < 0.2 ms to copy the cursor.

**Comment by cmeissl** [comment]
Waiting for https://github.com/Smithay/smithay/pull/1302

**Comment by YaLTeR** [comment]
Thanks! Yeah, now `render_cursor_plane()` has disappeared completely, save for three calls, for a similar test scenario to what I did above when it was in the hundreds. `try_assign_cursor_plane()`, whenever it is non-trivial, is in the 0.2-0.3 ms range (note linear time and log values):

image

There are just a few high outliers that seem to be some GPU stalls:

image

**Pull request merged**